x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -102,13 +102,15 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
             SSLEngineUtils.extractClientCertificates(logger, threadContext, httpChannel);
         }
 
-        authenticationService.authenticate(maybeWrapRestRequest(request), ActionListener.wrap(authentication -> {
+        final RestRequest wrappedRequest = maybeWrapRestRequest(request);
+        authenticationService.authenticate(wrappedRequest, ActionListener.wrap(authentication -> {
```

Comment:
This `authenticationService.authenticate` method is to be moved to a sooner site in the request handling flow (where it will not have access to the request body).